Web App - Deploy web app always wants to update app service plan

### Plugin name and version
azure-webapp-maven-plugin
1.7.0

### Plugin configuration in your `pom.xml`
            <plugin>
                <!-- see https://docs.microsoft.com/en-gb/java/azure/spring-framework/deploy-spring-boot-java-app-with-maven-plugin?toc=%2Fazure%2Fapp-service%2Fcontainers%2Ftoc.json&view=azure-java-stable -->
                <groupId>com.microsoft.azure</groupId>
                <artifactId>azure-webapp-maven-plugin</artifactId>
                <version>1.7.0</version>
                <configuration>
                    <deploymentType>jar</deploymentType>

                    <!-- configure app to run on port 80, required by App Service -->
                    <appSettings>
                        <property>
                            <name>JAVA_OPTS</name>
                            <value>-Dserver.port=80</value>
                        </property>
                    </appSettings>

                    <!-- Web App information -->
                    <resourceGroup>${azure.resourceGroup}</resourceGroup>
                    <appName>${azure.webappName}</appName>
                    <region>${azure.region}</region>

                    <!-- Java Runtime Stack for Web App on Linux-->
                    <linuxRuntime>java11</linuxRuntime>
                    <runtime>
                        <os>Linux</os>
                        <javaVersion>java11</javaVersion>
                        <webContainer></webContainer>
                    </runtime>

                    <stopAppDuringDeployment>true</stopAppDuringDeployment>
                </configuration>
            </plugin>

### Expected behavior
We just want to deploy an application to the web app service.
We don't want to update the service plan.
We don't want to create a new web app.

### Actual behavior
The plugin always tries to update the app service plan, and this causes an error.

Is there any way to flag to the plugin to not attempt to create or modify the service plan?

We are running in an environment with restricted permissions; we believe that we have enough permissions to be able to update the web app, but we don't have sufficient permissions to update the app service plan. (We may not even have permissions to _read_ the app service plan).

We get a cryptic error message :

[ERROR] Failed to execute goal com.microsoft.azure:azure-webapp-maven-plugin:1.7.0:deploy (default-cli) on project gps: WorkerSizeId is invalid.: OnError while emitting onNext value: retrofit2.Response.class -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.microsoft.azure:azure-webapp-maven-plugin:1.7.0:deploy (default-cli) on project gps: WorkerSizeId is invalid.


### Steps to reproduce the problem
$ mvn -X azure-webapp:deploy
...
[INFO] Updating app service plan
[INFO] --> PUT https://management.azure.com/subscriptions/93b40bf7-c109-4b1f-a0ff-8b594212a646/resourceGroups/ProdAppsRG/providers/Microsoft.Web/serverfarms/ASP-ProdAppsRG-bb35?api-version=2018-02
-01
[INFO] Accept-Encoding: gzip
[INFO] accept-language: en-US
[INFO] Authorization: Bearer  [SNIPPED]
3AquoK3t8JN--iB0ieew
[INFO] Connection: Keep-Alive
[INFO] Content-Length: 493
[INFO] Content-Type: application/json; charset=utf-8
[INFO] Host: management.azure.com
[INFO] User-Agent: azure-webapp-maven-plugin/1.7.0 installationId:0f85959563ec7eb53710917af9dd8f17c2fffcf62935322964ee43a947d508e1 sessionId:f2c09508-87c3-4278-9f28-20deb1c0afdb Azure-SDK-For-Java
/1.21.0 OS:Windows Server 2012 R2/6.3 MacAddressHash:4b25ffc4a6ddfdc5bf0f67a5980af9ae114264f788cf462ef2ac805faa20568a Java:11.0.4 (WebSiteManagementClient, 2018-02-01)
[INFO] x-ms-client-request-id: b4906af3-44a8-401a-8eae-39541601e715
[INFO] 493-byte body:
{"location":"UK South","tags":{"Location":"UK South","Environment":"Production","Application":"App Service Plan"},"sku":{"name":"P1v2","tier":"PremiumV2","size":"P1v2"},"kind":"linux","properties":{"hostingEnvironmentProfile":{"id":"/subscriptions/93b40bf7-c109-4b1f-a0ff-8b594212a646/resourceGroups/GrowWithGreencoreRG/providers/Microsoft.Web/hostingEnvironments/gcg-ase-01"},"perSiteScaling":false,"isSpot":false,"reserved":true,"isXenon":false,"targetWorkerCount":0,"targetWorkerSizeId":0}}
[INFO] --> END PUT
[INFO] <-- 400  https://management.azure.com/subscriptions/93b40bf7-c109-4b1f-a0ff-8b594212a646/resourceGroups/ProdAppsRG/providers/Microsoft.Web/serverfarms/ASP-ProdAppsRG-bb35?api-version=2018-0
2-01 (227 ms, 327-byte body)
[INFO] cache-control: no-cache
[INFO] content-length: 327
[INFO] content-type: application/json; charset=utf-8
[INFO] date: Thu, 05 Dec 2019 16:13:41 GMT
[INFO] expires: -1
[INFO] pragma: no-cache
[INFO] server: Microsoft-IIS/10.0
[INFO] strict-transport-security: max-age=31536000; includeSubDomains
[INFO] x-aspnet-version: 4.0.30319
[INFO] x-content-type-options: nosniff
[INFO] x-ms-correlation-request-id: d4bd3c63-f632-4eda-a4d6-c7b44ffa3b0c
[INFO] x-ms-ratelimit-remaining-subscription-writes: 1198
[INFO] x-ms-request-id: d4bd3c63-f632-4eda-a4d6-c7b44ffa3b0c
[INFO] x-ms-routing-request-id: UKWEST:20191205T161341Z:d4bd3c63-f632-4eda-a4d6-c7b44ffa3b0c
[INFO] x-powered-by: ASP.NET
[INFO] 327-byte body:
{"Code":"BadRequest","Message":"WorkerSizeId is invalid.","Target":null,"Details":[{"Message":"WorkerSizeId is invalid."},{"Code":"BadRequest"},{"ErrorEntity":{"ExtendedCode":"51012","MessageTempl
ate":"{0} is invalid.","Parameters":["WorkerSizeId"],"Code":"BadRequest","Message":"WorkerSizeId is invalid."}}],"Innererror":null}
[INFO] <-- END HTTP
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE

--- a/azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java
+++ b/azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java
@@ -119,8 +119,10 @@ public class DeployMojo extends AbstractWebAppMojo {
                 az.appServicePlan(getServicePlanResourceGroup(webAppConfig), webAppConfig.getServicePlanName());
         if (!targetServicePlan.exists()) {
             targetServicePlan = getOrCreateAppServicePlan(webAppConfig);
+        } else if (webAppConfig.getPricingTier() != null) {
+            targetServicePlan.update().withPricingTier(webAppConfig.getPricingTier()).commit();
         }
-        targetServicePlan.update().withPricingTier(webAppConfig.getPricingTier()).commit();
+
         final IWebApp result = webApp.update().withPlan(targetServicePlan.id())
                 .withRuntime(webAppConfig.getRuntime())
                 .withDockerConfiguration(webAppConfig.getDockerConfiguration())

